misc/bstr.c

```diff
@@ -454,3 +454,27 @@ struct bstr bstr_get_ext(struct bstr s)
         return (struct bstr){NULL, 0};
     return bstr_splice(s, dotpos + 1, s.len);
 }
+
+struct bstr bstr_decode_hex(void *talloc_ctx, struct bstr hex)
+{
+    char *out = talloc_array(talloc_ctx, char, hex.len);
+    int pos = 0;
+
+    // Read out all hex chars as individual nibbles
+    for (int i = 0; i < hex.len; i++) {
+        unsigned char in = hex.start[i];
+        if (in >= '0' && in <= '9')
+            out[pos++] = in - '0';
+        if (in >= 'a' && in <= 'f')
```

Comment:
The compiler would also generate the same code, so it's only a readability argument.